### PR TITLE
[MGDOBR-966] - Remove hardcoded default namespace reference

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
https://issues.redhat.com/browse/MGDOBR-966

The actual PR to remove hard coded namespace is #974
but PR 947 is failing because change in `operator-CI.yaml` will only take effect after they push. that's why I created this PR just for `operator-CI.yaml` changes. Compilation issues of #974 would be resolved once this PR is merged.

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by robpblake — COMMENTED** [review]
@vaibhavjainwiz Can you provide some more context on what this change will do? I don't see the removal of any hardcoded value at this point and the associated JIRA is closed as "won't do".

cheers,
Rob

**Comment by vaibhavjainwiz** [comment]
earlier I close that Jira by considering that these changes might not required
but I go further with manage-tenant onboarding, we need these changes. 
I reopened that Jira

File: .github/workflows/Operator-CI.yaml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
       - name: Build Shard operator image and resources
         uses: ./.github/actions/maven
         with:
-          maven-command: clean install --errors --quiet -Pminikube -Dquarkus.container-image.tag=$GITHUB_SHA -Dnamespace=$NAMESPACE -Dquickly
+          maven-command: clean install --errors --quiet -Pminikube -Dquarkus.container-image.tag=$GITHUB_SHA -Dnamespace=$NAMESPACE -Dquickly -Dquarkus.kubernetes.namespace=$NAMESPACE
```
Comment (r00ta) [inline .github/workflows/Operator-CI.yaml:53]: I still don't get how the Operator-CI is related to the manage-tenant onboarding, could you clarify?

**Comment by vaibhavjainwiz** [comment]
@robpblake @r00ta @danielezonca 
The actual PR to remove hard coded namespace is #974
but PR 947 is failing because change in `operator-CI.yaml` will only take effect after they push. that's why I created this PR just for `operator-CI.yaml` changes. Compilation issues of #974 would be resolved once this PR is merged.

**Review by danielezonca — APPROVED** [review]

File: .github/workflows/Operator-CI.yaml (hunk 1 above)
Comment (vaibhavjainwiz) [reply .github/workflows/Operator-CI.yaml:53]: Expectation
For manage-tenant on-boarding, generated operator-bundle should be independent of namespaces. Namespace value will be injected while creating addon.

Actual
With the current implementation, namespace attribute always injected in to Deployment and ServiceAccount while compiling project using maven.

Solution
Remove hardcoded default namespace reference from pom.xml and only provide namespace value while compiling project using
-Dquarkus.kubernetes.namespace=$NAMESPACE argument.

**Comment by robpblake** [comment]
OK thanks for the explanation @vaibhavjainwiz 👍

**Review by robpblake — APPROVED** [review]

**Pull request merged**